playground/backend/internal/utils/preparares_utils_test.go

```diff
@@ -50,3 +60,328 @@ func TestSpacesToEqualsOption(t *testing.T) {
+	}{
+		{
+			name: "Add graph to the end of an existing file",
+			args: args{
+				spaces:       "",
+				err:          nil,
+				tempFile:     txtFile,
+				pipelineName: uuid.New().String(),
+			},
+			fields: fields{
+				fileContent: fileContent,
+				filePath:    txtFilePath,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Error during write data to file",
+			args: args{
+				spaces:       "",
+				err:          nil,
+				tempFile:     incorrectFile,
+				pipelineName: uuid.New().String(),
+			},
+			fields: fields{
+				fileContent: fileContent,
+				filePath:    txtFilePath,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			AddGraphToEndOfFile(tt.args.spaces, tt.args.err, tt.args.tempFile, tt.args.pipelineName)
+			data, err := os.ReadFile(tt.fields.filePath)
+			if err != nil {
+				t.Errorf("AddGraphToEndOfFile() error during reading from a file = %v", err)
+			}
+			graphCode := fmt.Sprintf(pythonGraphCodePattern, tt.args.pipelineName, GraphFileName)
+			graphCodeWithIndentation := strings.ReplaceAll(graphCode, indentationReplacement, tt.args.spaces)
+			fileContentWithGraph := fileContent + "\n" + graphCodeWithIndentation
+			if (string(data) != fileContentWithGraph) != tt.wantErr {
+				t.Error("AddGraphToEndOfFile() wrong graph addition")
+			}
+		})
+	}
+}
+
+func TestGetPublicClassName(t *testing.T) {
+	javaPublicClassNamePattern := "public class (.*?) [{|implements(.*)]"
+	type args struct {
+		filePath string
+		pattern  string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "Get public class name from existing java file",
+			args: args{
+				filePath: filepath.Join(sourceDir, javaFileName),
+				pattern:  javaPublicClassNamePattern,
+			},
+			want:    "MinimalWordCount",
+			wantErr: false,
+		},
+		{
+			name: "Get public class name from non-existent file",
+			args: args{
+				filePath: filepath.Join(sourceDir, "file1.java"),
+				pattern:  javaPublicClassNamePattern,
+			},
+			want:    "",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetPublicClassName(tt.args.filePath, tt.args.pattern)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetPublicClassName() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("GetPublicClassName() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestAddNewLine(t *testing.T) {
+	javaFile, err := os.OpenFile(filepath.Join(sourceDir, javaFileName), os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err != nil {
+		panic(err)
+	}
+	defer javaFile.Close()
+	txtFile, err := os.Open(filepath.Join(sourceDir, fileName))
+	if err != nil {
+		panic(err)
+	}
+	defer txtFile.Close()
+	type args struct {
+		newLine bool
+		file    *os.File
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "With newLine = false",
```

Comment:
Can be renamed like "No line added to file"

Reply:
Done